### PR TITLE
Bicaws 285 - Create Kibana dashboard

### PR DESCRIPTION
**Pull request by TheGrantsta**
This will create a Kibana dashboard based on the exported configuration settings in the dashboard.json file.

The Lambda function is a zipped at the moment and I suggest creating another card for compiling Lambda functions on the fly (Kibana and Elasticsearch) rather than hold this card back any longer.

### CODE REVIEW

File: terraform_modules/kibana/main.tf [hunk 1]
```diff
@@ -66,47 +140,53 @@ resource "aws_iam_role_policy" "kibana_lambda_policy" {
...
-        "ec2:DescribeImages",
-        "es:ESHttpPost"
-    ],
-      "Resource": "*"
-    }
-  ]
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor1",
```
Comment (brettminnie) [inline terraform_modules/kibana/main.tf:146]: Give this a more friendly name please

File: terraform_modules/kibana/main.tf [hunk 2]
```diff
@@ -142,10 +211,29 @@ resource "aws_lambda_permission" "allow_s3" {
   depends_on      = [aws_iam_role_policy.kibana_lambda_policy]
 }
 
+resource "aws_s3_bucket_notification" "s3-trigger" {
```
Comment (brettminnie) [inline terraform_modules/kibana/main.tf:214]: rename to `s3_trigger`

File: terraform/stack_infra_baseline/main.tf [hunk 3]
```diff
@@ -186,13 +186,13 @@ module "elastic_search" {
...
-  source             = "../../terraform_modules/kibana"
-  region_name        = data.aws_region.current_region.name
-  name               = module.label.id
-  tags               = module.label.tags
-  kibana_endpoint    = module.elastic_search.elk_endpoint
-  vpc_id             = data.terraform_remote_state.base_infra.outputs.vpc.vpc_id
-  admin_allowed_cidr = data.terraform_remote_state.base_infra.outputs.vpc.private_subnet_cidrs
-  service_subnets    = data.terraform_remote_state.base_infra.outputs.vpc.private_subnet_ids
-  bucket_name        = data.terraform_remote_state.base_infra.config.bucket
+  source                  = "../../terraform_modules/kibana"
```
Comment (brettminnie) [inline terraform/stack_infra_baseline/main.tf:189]: I'd like to see if we can replace this with `"${path.root}/terraform_modules/kibana"`

Comment (TheGrantsta) [reply terraform/stack_infra_baseline/main.tf:189]: This change seemed to cause an error: 
Error: Variables not allowed

  on main.tf line 189, in module "kibana":
 189:   source                  = "${path.root}/terraform_modules/kibana"

Variables may not be used here.

File: terraform_modules/kibana/main.tf [hunk 4]
```diff
@@ -143,7 +143,7 @@ resource "aws_iam_role_policy" "kibana_lambda_policy" {
     "Version": "2012-10-17",
     "Statement": [
         {
-            "Sid": "VisualEditor1",
```
Comment (brettminnie) [inline terraform_modules/kibana/main.tf]: We've used a camel case convention up until now, lets keep it consistent

**Review by brettminnie — CHANGES_REQUESTED** [review]

**Review by brettminnie — APPROVED** [review]

**Pull request merged**